test/SlowTests/Cluster/ShardedClusterTransactionTests.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using FastTests.Client;
+using FastTests.Server.Replication;
+using Raven.Client.Documents;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.Documents.Session;
+using Raven.Client.ServerWide;
+using Tests.Infrastructure;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Cluster
+{
+    public class ShardedClusterTransactionTests : ClusterTransactionTests
+    {
+        public ShardedClusterTransactionTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Theory]
+        [InlineData(1, 1, false)]
+        [InlineData(2, 1, false)]
+        [InlineData(1, 2, false)]
+        [InlineData(2, 2, false)]
+        [InlineData(1, 1, true)]
+        [InlineData(2, 1, true)]
+        [InlineData(1, 2, true)]
+        [InlineData(2, 2, true)]
+        public async Task ShardedClusterTransaction_ChangeVector(int numberOfNodes, int numberOfShards, bool disableAtomicGuard)
+        {
+            var (_, leader) = await CreateRaftCluster(numberOfNodes, false, watcherCluster: true);
+
+            using var store = GetDocumentStore(new Options
+            {
+                Server = leader,
+                ModifyDatabaseRecord = r => 
+                    r.Shards = Enumerable.Range(0, numberOfShards).Select(_ => new DatabaseTopology()).ToArray(),
+                ReplicationFactor = 2
+            });
+
+            using (var session = store.OpenAsyncSession(new SessionOptions
+            {
+                TransactionMode = TransactionMode.ClusterWide,
+                DisableAtomicDocumentWritesInClusterWideTransaction = disableAtomicGuard
+            }))
+            {
+                var entities = new List<TestObj>();
+                for (int i = 0; i < 10; i++)
+                {
+                    var testObj = new TestObj();
+                    entities.Add(testObj);
+                    await session.StoreAsync(testObj, $"TestObjs/{i}");
+                }
+                
+                await session.SaveChangesAsync();
+
+                // WaitForUserToContinueTheTest(store);
+
+                //TODO To remove waiting
+                await Task.Delay(2000);
+                using var session2 = store.OpenAsyncSession(new SessionOptions
```

Comment:
For now the `Load` does not wait on the raft index

Reply:
what do you mean?

Reply:
As I understand it, for now, we don't read `Constants.Headers.LastKnownClusterTransactionIndex` in a shard database handler (in the `RequestRouter`).
So the client doesn't wait for the cluster transaction to be executed and the test got null here.

Reply:
Why aren't we just forwarding the value to the shard instances? Then it will Just Work, no?

Reply:
We need to add that to the header of the requests sent from the shard handler.
I can add it here for `Load` but I guess it will be more appropriate to implement something more generic.
I will start to check that but maybe it is worth separating it to another issue.

Reply:
That should be passed along always, yes. Can add that in the shard handler base class, I think